framework/include/base/MooseObjectStorage.h

```diff
@@ -0,0 +1,398 @@
+        _active_block_objects[tid][*it].push_back(object);
+    }
+  }
+}
+
+
+template<typename MooseObjectType>
+bool
+MooseObjectStorage<MooseObjectType>::hasActiveBlockObjects(THREAD_ID tid/* = 0*/) const
+{
+  checkThreadID(tid);
+  bool has_active_block_objects = false;
+  typename std::map<SubdomainID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator it;
+  for (it = _active_block_objects[tid].begin(); it != _active_block_objects[tid].end(); ++it)
+    has_active_block_objects |= !(it->second.empty());
+  return has_active_block_objects;
+}
+
+
+template<typename MooseObjectType>
+bool
+MooseObjectStorage<MooseObjectType>::hasActiveBlockObjects(SubdomainID id, THREAD_ID tid/* = 0*/) const
+{
+  checkThreadID(tid);
+  typename std::map<SubdomainID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator iter = _active_block_objects[tid].find(id);
+  if (iter != _active_block_objects[tid].end() && !iter->second.empty())
+    return true;
+  return false;
+}
+
+
+template<typename MooseObjectType>
+bool
+MooseObjectStorage<MooseObjectType>::hasActiveBoundaryObjects(THREAD_ID tid/* = 0*/) const
+{
+  checkThreadID(tid);
+  bool has_active_boundary_objects = false;
+  typename std::map<BoundaryID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator it;
+  for (it = _active_boundary_objects[tid].begin(); it != _active_boundary_objects[tid].end(); ++it)
+    has_active_boundary_objects |= !(it->second.empty());
+  return has_active_boundary_objects;
+}
+
+
+template<typename MooseObjectType>
+bool
+MooseObjectStorage<MooseObjectType>::hasActiveBoundaryObjects(BoundaryID id, THREAD_ID tid/* = 0*/) const
+{
+  checkThreadID(tid);
+  typename std::map<BoundaryID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator iter = _active_boundary_objects[tid].find(id);
+  if (iter != _active_boundary_objects[tid].end() && !iter->second.empty())
+    return true;
+  return false;
+}
+
+
+template<typename MooseObjectType>
+void
+MooseObjectStorage<MooseObjectType>::updateActive(THREAD_ID tid /*= 0*/)
+{
+  checkThreadID(tid);
+
+  updateActiveHelper(_active_objects[tid], _all_objects[tid]);
+
+  {
+    typename std::map<SubdomainID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator it;
+    for (it = _all_block_objects[tid].begin(); it != _all_block_objects[tid].end(); ++it)
+      updateActiveHelper(_active_block_objects[tid][it->first], it->second);
+  }
+
+
+  {
+    typename std::map<BoundaryID, std::vector<MooseSharedPointer<MooseObjectType> > >::const_iterator it;
+    for (it = _all_boundary_objects[tid].begin(); it != _all_boundary_objects[tid].end(); ++it)
+      updateActiveHelper(_active_boundary_objects[tid][it->first], it->second);
+  }
+
+}
+
+
+template<typename MooseObjectType>
+void
+MooseObjectStorage<MooseObjectType>::updateActiveHelper(std::vector<MooseSharedPointer<MooseObjectType> > & active, const std::vector<MooseSharedPointer<MooseObjectType> > & all)
+{
```

Comment:
Holy Dooley: `isActive` is a beast as well to be calling on each element and side:
```
bool
TransientInterface::isActive()
{
  // if we have zero or one time periods -> all objects are active all the time
  if (_time_periods.size() <= 1)
    return true;

  // look if _t lies in one of our time periods
  for (unsigned int i=1; i <= _time_periods.size(); ++i)  // Careful! We are purposely indexing one past the end of the array
    if ((i == _time_periods.size()) || // Are we in the last time period?
        ((_time_periods[i-1]->start() <= _t) && (_t < _time_periods[i]->start())))  // OR are we in one of the intermediate periods?
    {
      bool ret_value;
      const std::vector<std::string> & objects = _time_periods[i-1]->getObjectList(_object_type, ret_value);
      if (std::find(objects.begin(), objects.end(), _ti_name) != objects.end())
        return ret_value;
      else
        return !ret_value;
    }

  return false;
}
```